src/lib/generate-matcher.js

```diff
@@ -72,6 +73,23 @@ const getParamsMatcher = ({ params: expectedParams, matcher }) => {
 const getFunctionMatcher = ({ matcher, functionMatcher = () => true }) =>
 	typeof matcher === 'function' ? matcher : functionMatcher;
 
+const getBodyMatcher = ({ body: expectedBody }) => {
+	return (url, { headers = {}, body, method = 'get' }) => {
+		if (method.toLowerCase() === 'get') {
+			// GET requests don’t send a body so the body matcher should be ignored for them
+			return true;
+		}
+
+		const lowerCaseHeaders = headerUtils.toLowerCase(
+			headerUtils.normalize(headers)
+		);
+
+		return lowerCaseHeaders['content-type'] === 'application/json' && body
```

Comment:
I wonder if this is a good idea to check for this. Even though it's bad practice to, there are circumstances where applications are sending json bodies without appropriate headers, and users could quite legitimately want to test those scenarios. Would there be any downsides to loosening this up to not require the header?

Reply:
Having thought about it, I think you’re right, I think I’ll remove that requirement